lib/llm/src/http/service/health.rs

```diff
@@ -0,0 +1,167 @@
+// SPDX-FileCopyrightText: Copyright (c) 2024-2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// SPDX-FileCopyrightText: Copyright (c) 2024-2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::{service_v2, RouteDoc};
+use axum::{
+    extract::Path, http::Method, http::StatusCode, response::IntoResponse, routing::get, Json,
+    Router,
+};
+use dynamo_runtime::component::{Instance, INSTANCE_ROOT_PATH};
+use dynamo_runtime::{DistributedRuntime, Runtime};
+use serde_json::json;
+use std::sync::Arc;
+
+pub fn health_check_router(
+    state: Arc<service_v2::State>,
+    path_override: Option<String>,
+) -> (Vec<RouteDoc>, Router) {
+    let path = path_override.unwrap_or_else(|| "/health".to_string());
+    let path_namespace = format!("{path}/{{namespace}}");
+    let path_component = format!("{path}/{{namespace}}/{{component}}");
+
+    let docs: Vec<RouteDoc> = vec![
+        RouteDoc::new(Method::GET, &path),
+        RouteDoc::new(Method::GET, &path_namespace),
+        RouteDoc::new(Method::GET, &path_component),
+    ];
+
+    let router = Router::new()
+        .route(&path, get(health_handler))
+        .route(&path_namespace, get(health_namespace_handler))
+        .route(&path_component, get(health_component_handler))
+        .with_state(state);
+
+    (docs, router)
+}
+
+async fn health_handler() -> impl IntoResponse {
+    Json(json!({"status": "healthy"}))
+}
+
+// A namespace health check will return if the namespace exists in ETCD and will return a list of components currently registered
+async fn health_namespace_handler(Path(namespace): Path<String>) -> impl IntoResponse {
+    let target_key = format!("{INSTANCE_ROOT_PATH}/{namespace}");
```

Comment:
The HTTP handlers shouldn't know etcd path internals, and shouldn't go to etcd directly. They get a `state` parameter which should have what they need. Currently it has the `ModelManager` which allows you to access *models* but not the components and instances serving the model.

I think you need either `Component.instances()` or `ModelWatcher.all_entries()`, neither of which is currently accessible from the `State` object. Let me fix that.